src/Containers/SlidingOutPanels.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { MainInterface, ResponsiveInterface } from '../Utils/BaseStyles';
+
+interface ImageProps {
+    id: string;
+    imageSource: string;
+    alt: string;
+}
+
+export interface SlidingOutPanelProps
+    extends MainInterface,
+        ResponsiveInterface {
+    images: ImageProps[];
+}
+
+export const SlidingOutPanels: React.FC<SlidingOutPanelProps> = ({
+    images,
+}): React.ReactElement => {
+    return (
+        <Items>
+            <Overlay>
+                {images.map(
+                    ({ id, imageSource, alt }): React.ReactElement => (
+                        <Image key={id} id={id} src={imageSource} alt={alt} />
+                    ),
+                )}
+            </Overlay>
+        </Items>
+    );
+};
+
+const Items = styled.div`
+    width: 500px;
+    height: 350px;
+    padding: 30px;
+    box-sizing: border-box;
+    overflow: hidden;
+`;
+
+const Overlay = styled.div`
+    width: 450px;
+    height: 200px;
+    position: relative;
+    border-radius: 15px;
+`;
+
+interface IdProps {
```

Comment:
Why name this IdProps instead of ImageProps? What happens if you add another prop to the ID component?

Reply:
I did go for ImageProps initially but decided to use a new interface IdProps because it was conflicting with the TS type of the image component returned by the react function. I have now switched back to ImageProps and handled the conflict by making the uncommon TS types optional. Sorry about the churn, just getting a feel with Typescript. This change has been made.

Reply:
Lets discuss on a call. There’s something I want to explain with Styled Components

Reply:
reverted back to a separate props interface with a better naming convention